nativescript-angular/router/ns-location-strategy.ts

```diff
@@ -239,7 +240,14 @@ export class NSLocationStrategy extends LocationStrategy {
 
     private callPopState(state: LocationState, pop: boolean = true) {
         const urlSerializer = new DefaultUrlSerializer();
-        this.currentUrlTree.root.children[this.currentOutlet] = state.segmentGroup;
+        if (state) {
+            this.currentUrlTree.root.children[this.currentOutlet] = state.segmentGroup;
+        } else {
```

Comment:
Suggestion: add a comment on when `state` comes in as `undefined` and why  the cleanup logic is needed.